allel/io/vcf_read.py

```diff
@@ -303,13 +307,13 @@ def read_vcf(input,
     # read all chunks into a list
     chunks = [d[0] for d in it]
 
-    # setup output
-    output = dict()
+    if chunks:
 
-    if len(samples) > 0 and store_samples:
-        output['samples'] = samples
+        # setup output
+        output = dict()
```

Comment:
Does it seem more consistent to return an empty dict than `None`? 
Although then you can have a problem with whether to store `samples`

Reply:
Yeah, I went back and forth on that one. In the end I decided it would be clearer and simpler to just return None if there are no variants.